cmd/ocm/create/cluster/cmd.go

```diff
@@ -180,58 +182,104 @@ func run(cmd *cobra.Command, argv []string) error {
+	err = arguments.PromptOrCheckOneOf(cmd.Flags(), "provider", osdProviders())
+	if err != nil {
+		return err
+	}
+
+	err = arguments.PromptBool(cmd.Flags(), "ccs")
+	if err != nil {
+		return err
+	}
+	if args.ccs.Enabled {
+		err = arguments.PromptString(cmd.Flags(), "aws-access-key-id")
+		if err != nil {
+			return err
+		}
+
+		err = arguments.PromptPassword(cmd.Flags(), "aws-secret-access-key")
+		if err != nil {
+			return err
+		}
+	}
+	err = arguments.CheckIgnoredCCSFlags(args.ccs)
+	if err != nil {
+		return err
+	}
+
+	regionSet := sets.NewString()
+	regions, err := provider.GetRegions(connection.ClustersMgmt().V1(), args.provider, args.ccs)
+	if err != nil {
+		return err
+	}
+	for _, region := range regions {
+		regionSet.Insert(region.ID())
+	}
+	arguments.PromptOrCheckOneOf(cmd.Flags(), "region", regionSet.List())
+
+	if args.region == "us-east-1" && args.provider != "aws" {
+		return fmt.Errorf("if specifying a non-aws cloud provider, region must be set to a valid region")
 	}
-	clusterName := argv[0]
 
-	// Validate flags:
 	expiration, err := c.ValidateClusterExpiration(args.expirationTime, args.expirationSeconds)
 	if err != nil {
 		return fmt.Errorf(fmt.Sprintf("%s", err))
 	}
-	versionList := sets.NewString()
+
+	versionSet := sets.NewString()
 	versions, err := fetchEnabledVersions(cmv1Client)
 	if err != nil {
 		return fmt.Errorf("unable to retrieve versions: %s", err)
 	}
 	for _, version := range versions {
-		versionList.Insert(version.ID())
-
+		// strip leading "openshift-v" string
+		versionSet.Insert(strings.Replace(version.ID(), "openshift-v", "", 1))
 	}
+	arguments.PromptOrCheckOneOf(cmd.Flags(), "version", versionSet.List())
 
 	// Check and set the cluster version
-	var clusterVersion string
 	if args.version != "" {
-		if strings.HasPrefix(args.version, "openshift-v") {
-			clusterVersion = args.version
-		} else {
-			clusterVersion = "openshift-v" + args.version
-		}
-		if !versionList.Has(clusterVersion) {
-			return fmt.Errorf("A valid version number must be specified\nValid versions: %+v", versionList.List())
+		if !strings.HasPrefix(args.version, "openshift-v") {
+			args.version = "openshift-v" + args.version
 		}
 	}
 
 	// Retrieve valid/default flavours
-	flavourList := sets.NewString()
+	flavourSet := sets.NewString()
 	flavours, err := fetchFlavours(cmv1Client)
 	if err != nil {
 		return fmt.Errorf("unable to retrieve flavours: %s", err)
 	}
 	for _, flavour := range flavours {
-		flavourList.Insert(flavour.ID())
+		flavourSet.Insert(flavour.ID())
 	}
 
-	// Check and set the cluster flavour
-	var clusterFlavour string
 	if args.flavour != "" {
-
-		if !flavourList.Has(args.flavour) {
-			return fmt.Errorf("A valid flavour number must be specified\nValid flavours: %+v", flavourList.List())
+		// TODO: user will have no idea if they want '4' vs 'osd-4' :-(
+		err = arguments.PromptOrCheckOneOf(cmd.Flags(), "flavour", flavourSet.List())
```

Comment:
I don't think we want to expose flavours. I think these should stay as _just_ flags: if the user knows what they want they can explicitly set `--flavour`, otherwise it should always be empty and let CS decide for them.

Reply:
Dropped the prompt, kept the validation.